File: 1.两数之和.cs
/*
 * @lc app=leetcode.cn id=1 lang=csharp
 *
 * [1] 两数之和
 */

// @lc code=start
public partial class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        for (int currentIndex = 0; currentIndex < nums.Length; currentIndex++)
        {
            int current = nums[currentIndex];
            int search = target - current;
            if (dict.TryGetValue(search, out int searchIndex))
            {
                return new int[] { searchIndex, currentIndex };
            }
            dict.TryAdd(current, currentIndex);
        }
        return Array.Empty<int>();
    }
}
// @lc code=end

File: src/1.两数之和.rs

/*
 * @lc app=leetcode.cn id=1 lang=rust
 *
 * [1] 两数之和
 */
// @lc code=start
use std::{collections::HashMap, vec};
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hashMap = HashMap::<i32, i32>::new();
        for (current_index, &current_numm) in nums.iter().enumerate() {
            let current_index = current_index as i32;
            let search_num = target - current_numm;
            if let Some(&search_index) = hashMap.get(&search_num) {
                return vec![search_index, current_index];
            } else {
                hashMap.entry(current_numm).or_insert(current_index);
            }
        }
        return vec![];
    }
}
// @lc code=end
